packages/react-native-analytics/src/integrations/castle/Castle.js

```diff
@@ -0,0 +1,229 @@
+import { Integration } from '@farfetch/blackout-core/analytics/integrations';
+import coreClient from '@farfetch/blackout-core/helpers/client';
+
+export const LOGGER_MESSAGE_PREFIX = 'Castle:';
+export const HTTP_CLIENT_ERROR_MESSAGE = `${LOGGER_MESSAGE_PREFIX} Make sure you are passing your HTTP client with the correct header. We strongly advise to use an interceptor to correctly set the Castle.io header as it should be treated as a token instead of a static value.`;
+
+class Castle extends Integration {
+  /**
+   * Method to check if the integration is ready to be loaded.
+   * Since this is a mandatory integration, it will always return true and will not depend on user consent.
+   * @static
+   *
+   * @returns {Boolean} - If the integration is ready to be loaded.
+   */
+  static shouldLoad() {
+    return true;
+  }
+
+  /**
+   * Method used to create a new Castle instance by analytics.
+   *
+   * @static
+   *
+   * @param {object} options - Integration options.
+   * @param {object} loadData - analytics' load event data.
+   * @param {object} strippedAnalytics - analytics' stripped instance methods.
+   *
+   * @returns {object} - An instance of Castle class.
+   */
+  static createInstance(options, loadData, strippedAnalytics) {
+    return new Castle(options, loadData, strippedAnalytics);
+  }
+
+  /**
+   * Creates an instance of Castle.
+   *
+   * @param {object} options - User configured options.
+   * @param {object} loadData - analytics' load event data.
+   * @param {object} strippedAnalytics - analytics' stripped instance methods.
+   */
+  constructor(options, loadData, strippedAnalytics) {
+    super(options, loadData, strippedAnalytics);
+
+    // This will allow users to have access to the Castle.io module and perform any project-specific operations with it,
+    // by accessing the integration instance via `analytics.integration('castle').castleIO;`
+    this.castleIO = CastleReactNative;
+
+    this.lastUserId = null;
+    this.isInterceptorAttached = false;
+    this.httpClientInterceptor = null;
+    this.httpClient = options?.httpClient || coreClient;
+
+    this.initializePromiseResolve = null;
+    this.initializePromise = new Promise(initializePromiseResolve => {
+      this.initializePromiseResolve = initializePromiseResolve;
+    });
+
+    this.initialize(options);
+  }
+
+  /**
+   * Initialization method that will configure the Castle instance with the options provided to the integration.
+   * Calls the configureHttpClientHeader to configure the HTTP client to be used on the project that will be profiled by this integration.
+   *
+   * @async
+   *
+   * @param {object} options - User configured options.
+   */
+  async initialize(options) {
+    const safeOptions = options || {};
+
+    await this.configureHttpClientHeader(options);
+
+    try {
+      await this.castleIO
+        .configure({
+          publishableKey: safeOptions.publishableKey,
```

Comment:
Are some of these options mandatory for the castleIO `configure` method? If so, I think a validation should be done on those options values before calling `configure`, so that we can present a more informative error message. What do you think?